terraform/modules/paas/variables.tf

```
@@ -25,21 +25,17 @@ variable docker_credentials { type = map }
 
 variable app_secrets_variable { type = map } #secrets from yml file
 
+variable app_config_variable { type = map } #from yml file
+
 locals {
   postgres_service_name = "register-postgres-${var.app_environment}"
   redis_service_name    = "register-redis-${var.app_environment}"
   web_app_name          = "register-${var.app_environment}"
   web_app_start_command = "bundle exec rails db:migrate && bundle exec rails server -b 0.0.0.0"
-  app_environment_none_secrets = {
-    AUTHORISED_HOSTS               = "${local.web_app_name}.london.cloudapps.digital"
-    RAILS_ENV                      = "production"
-    RACK_ENV                       = "production"
-    RAILS_SERVE_STATIC_FILES       = true
-    SECRET_KEY_BASE                = "TestKey"
-  }
+  app_environment_config = var.app_config_variable
   app_environment_secrets = var.app_secrets_variable
 
-  app_environment = merge(local.app_environment_none_secrets, local.app_environment_secrets)
+  app_environment = merge(local.app_environment_config, local.app_environment_secrets)
```

```suggestion
  app_environment = merge(var.app_environment_config, local.app_environment_secrets)
```
if not merging in the root module, you could remove `local.app_environment_config` and just use `var`

---

yes that could work too...